File: 백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.py
import sys
# sys.stdin = open("input.txt", 'r')
input = sys.stdin.readline

N, K = map(int, input().split())
cnt = 0
check = [False] * (N+1)
ans = []
for i in range(2, N+1):
    if check[i] == False:
        check[i] = True
        ans.append(i)
        for j in range(i+i, N+1, i):
            if check[j] == False:
                check[j] = True
                ans.append(j)
print(ans[K-1])

File: 백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.js
let fs = require("fs")
let input = fs.readFileSync("/dev/stdin").toString().trim().split(" ")

const [N, K] = input.map(x => parseInt(x))
let arr = new Array(N+1).fill(1)
let ans = []

for (let i = 2; i <= N; i++ ) {
    if (arr[i] === 1) {
        arr[i] = 0
        ans.push(i)
        for (let j = i+i; j <= N; j += i) {
            if (arr[j] === 1) {
                arr[j] = 0
                ans.push(j)
            }
        }
    }
}
console.log(ans[K-1])